plugin/pkg/shoot/dns/admission.go

```diff
@@ -201,27 +207,43 @@ func (d *DNS) Admit(ctx context.Context, a admission.Attributes, o admission.Obj
 	return nil
 }
 
-func managePrimaryDNSProvider(dns *core.DNS, defaultDomains []string) error {
-	var usesDefaultDomain bool
-	for _, domain := range defaultDomains {
-		if strings.HasSuffix(*dns.Domain, "."+domain) {
-			usesDefaultDomain = true
-			break
-		}
+func checkPrimaryDNSProvider(dns *core.DNS, defaultDomains []string) error {
+	if dns == nil || dns.Domain == nil {
```

Comment:
Just a small improvement:

```suggestion
	if dns == nil || dns.Domain == nil || len(dns.Providers) == 0 {
```